### PR TITLE
ipn/ipnlocal, wgengine/magicsock: check Expired bool on Node; print error in Ping

### PR DESCRIPTION
**Pull request by andrew-d**
Change-Id: Ic5f533f175a6e1bb73d4957d8c3f970add42e82e

### CODE REVIEW

File: wgengine/magicsock/magicsock.go [hunk 1]
```diff
@@ -3647,6 +3647,9 @@ type endpoint struct {
 	// See #540 for background.
 	heartbeatDisabled bool
 	pathFinderRunning bool
+
+	// whether the node has expired
```
Comment (bradfitz) [inline wgengine/magicsock/magicsock.go:3651]: If it's a before-line comment, it should be a complete sentence "// expired is whether .."

But move it down to a same-line comment and this is fine

File: ipn/ipnlocal/local.go [hunk 2]
```diff
@@ -2275,6 +2275,9 @@ func (b *LocalBackend) pingPeerAPI(ctx context.Context, ip netip.Addr) (peer *ta
 	if !ok {
 		return nil, "", fmt.Errorf("no peer found with Tailscale IP %v", ip)
 	}
+	if peer.Expired {
+		return nil, "", errors.New("peer has expired")
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:2279]: "peer's key has expired" or something?

this is user-visible, so maybe @mayakacz has opinions. You can submit now and tweak later, but heads up to Maya.

File: wgengine/magicsock/magicsock.go [hunk 3]
```diff
@@ -3920,12 +3929,20 @@ func (de *endpoint) cliPing(res *ipnstate.PingResult, cb func(*ipnstate.PingResu
...
+)
+
 func (de *endpoint) send(buffs [][]byte) error {
 	if fn := de.sendFunc.Load(); fn != nil {
 		return fn(buffs)
 	}
 
 	de.mu.Lock()
+	if de.expired {
+		return errExpired
```
Comment (bradfitz) [inline wgengine/magicsock/magicsock.go:3944]: you didn't unlock de.mu

**Review by bradfitz — APPROVED** [review]

File: ipn/ipnlocal/local.go (hunk 2 above)
Comment (mayakacz) [reply ipn/ipnlocal/local.go:2279]: "peer's node key has expired" or "peer's node key is expired"

File: wgengine/magicsock/magicsock.go (hunk 3 above)
Comment (andrew-d) [reply wgengine/magicsock/magicsock.go:3944]: 🤦

File: ipn/ipnlocal/local.go (hunk 2 above)
Comment (andrew-d) [reply ipn/ipnlocal/local.go]: Done, thanks @mayakacz!

**Pull request merged**